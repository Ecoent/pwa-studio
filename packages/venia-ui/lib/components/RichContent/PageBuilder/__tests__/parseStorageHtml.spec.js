import parseStorageHtml from '../parseStorageHtml';

test('parse valid root element', () => {
    const testMasterFormat =
        '<div data-content-type="row" data-appearance="contained" data-element="main"><div class="test-class" data-enable-parallax="0" data-parallax-speed="0.5" data-background-images="{}" data-element="inner" style="justify-content: center; display: flex; flex-direction: column; background-color: rgb(17, 85, 0); background-position: left top; background-size: cover; background-repeat: no-repeat; background-attachment: scroll; text-align: center; border-style: solid; border-color: rgb(255, 0, 0); border-width: 10px; border-radius: 5px; min-height: 100px; margin: 5px; padding: 15px;"><h2 data-content-type="heading" data-appearance="default" data-element="main" style="border-style: none; border-width: 1px; border-radius: 0px;">Test Heading</h2></div></div>';

    const data = parseStorageHtml(testMasterFormat);
    expect(data).toMatchSnapshot();
});

test('test invalid content type and expect warning', () => {
    const testInvalidMasterFormat =
        '<div data-content-type="garbage"/>';

    jest.spyOn(console, 'warn').mockImplementation(() => {});
    parseStorageHtml(testInvalidMasterFormat);

    expect(console.warn).toHaveBeenCalled();
});
