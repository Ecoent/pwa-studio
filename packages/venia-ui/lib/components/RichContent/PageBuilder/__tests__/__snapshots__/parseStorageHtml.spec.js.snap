// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parse valid root element 1`] = `
Object {
  "appearance": null,
  "children": Array [
    Object {
      "appearance": "contained",
      "backgroundAttachment": "scroll",
      "backgroundColor": "rgb(17, 85, 0)",
      "backgroundPosition": "left top",
      "backgroundRepeat": false,
      "backgroundSize": "cover",
      "border": "solid",
      "borderColor": "rgb(255, 0, 0)",
      "borderRadius": "5px",
      "borderWidth": "10px",
      "children": Array [
        Object {
          "appearance": "default",
          "border": "none",
          "borderColor": "",
          "borderRadius": "0px",
          "borderWidth": "1px",
          "children": Array [],
          "contentType": "heading",
          "cssClasses": Array [],
          "headingType": "h2",
          "marginBottom": "",
          "marginLeft": "",
          "marginRight": "",
          "marginTop": "",
          "paddingBottom": "",
          "paddingLeft": "",
          "paddingRight": "",
          "paddingTop": "",
          "text": "Test Heading",
          "textAlign": "",
        },
      ],
      "contentType": "row",
      "cssClasses": Array [
        "test-class",
      ],
      "desktopImage": null,
      "enableParallax": false,
      "marginBottom": "5px",
      "marginLeft": "5px",
      "marginRight": "5px",
      "marginTop": "5px",
      "minHeight": "100px",
      "mobileImage": null,
      "paddingBottom": "15px",
      "paddingLeft": "15px",
      "paddingRight": "15px",
      "paddingTop": "15px",
      "parallaxSpeed": 0.5,
      "textAlign": "center",
      "verticalAlignment": "middle",
    },
  ],
  "contentType": "root-container",
}
`;
